pkg/session_recording/player.jsx

```diff
@@ -620,6 +639,8 @@
 
                     /* Skip packets we don't output */
                     if (!pkt.is_io || !pkt.is_output) {
+                        this.state.term.resize(pkt.width, pkt.height);
+                        this._transform(pkt.width, pkt.height);
```

Comment:
This is also being called on "input" packets, not only on "window" packets. You're not hitting this now, because input is not recorded by default. Please fix.

Reply:
Essentially, instead of adding code here, the `if` condition should be changed to `pkt.is_io && !pkt.is_output`, and the code added instead after the "Output the packet" comment below, branching on output/window packet type. This would also fix the timing issues with window resizing.

Reply:
I've made a bit different change, not sure if it's okay, but my implementation of your provided idea did not work, let's also discuss this tomorrow, probably I'm missing something.